### PR TITLE
Switch communication from createEvent to postMessage

### PR DESCRIPTION
**Pull request by zeckdude**
## Description

Firefox doesn't allow communicate with the content script via dispatchEvent so we need use `window.postMessage` instead.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#communicating_with_the_web_page

## Testing

- [ ] Includes test.
- [x] Manually tested.

## Additional notes

I didn't actually add any tests but after the changes the current tests still works.

### CODE REVIEW

**Comment by jcblw** [comment]
Still needs verification on Safari if we can use this method exclusively. I am assuming yes, but we really need to test it out.

```
finch-graphql@1.2.1-alpha
```

Published this version to do this testing.

**Comment by jcblw** [comment]
> Still needs verification on Safari if we can use this method exclusively. I am assuming yes, but we really need to test it out.
> 
> ```
> finch-graphql@1.2.1-alpha
> ```
> 
> Published this version to do this testing.

Verified this does in fact work on Safari, as well. Think we should eventually should rename, fear that is some breaking changes though.

File: packages/finch-graphql/src/external/listenForDocumentQueries.ts [hunk 1]
```diff
@@ -34,19 +39,24 @@ export const sendResponse = (
...
-    const resp = await queryApi(event.detail.query, event.detail.variables, {
+    const resp = await queryApi(event.data.query, event.data.variables, {
       ...options,
       external: true,
     });
-    sendResponse(event, resp.data, resp.errors);
+    sendResponse(event.data.requestId, resp.data, resp.errors);
   };
-  document.addEventListener(FinchDocumentEventNames.Request, onMessage);
+  // document.addEventListener(FinchDocumentEventNames.Request, onMessage);
```
Comment (jcblw) [inline packages/finch-graphql/src/external/listenForDocumentQueries.ts:55]: Can we remove this line.

**Review by jcblw — CHANGES_REQUESTED** [review]
One super minor change, just remove the commented out code and I can make a release for this.

**Comment by zeckdude** [comment]
I have removed the line. Thanks @jcblw

**Review by jcblw — APPROVED** [review]

**Pull request merged**